R/QCplots.R

```diff
@@ -71,160 +43,288 @@
+        warning("plotCategory must be one of 'bar', 'point', 'pointline', or 'histogram'. Either one category or as many categories as the number of metric names can be specified. Assigning default value 'bar'.")
+        plotCategory <- "bar"
+    }
+
+    #hlineSD
+    if (is.null(hlineSD)) {
+        hlineSD <- 0
+    } else if (!is.null(hlineSD) &&
+              any(!is.numeric(hlineSD),
+                  length(hlineSD) != 1)) {
+        warning("hlineSD needs to be a single numeric value. Assigning default value 3.")
+        hlineSD <- 3
+    }
+
+    #labelAngle
+    if (any(is.null(labelAngle),
+            !is.numeric(labelAngle),
+            length(labelAngle)  != 1)) {
+        warning("labelAngle must be a numeric value. Assigning default values 30.")
+        labelAngle <- 30
+    }
+
+    #winsorize
+    if (any(is.null(winsorize),
+            length(winsorize) != 1,
+            !is.logical(winsorize))) {
+        warning("winsorize must be a singular logical value. Assigning default value TRUE.")
+        winsorize <- TRUE
+    }
 
     # Replace "." in column 2-n with NA and convert columns 2-n to numeric
     dotIdx <- qcdata == "."
     if (sum(dotIdx, na.rm = TRUE) > 0) {
         qcdata[dotIdx] <- NA
+        qcdata <- dplyr::mutate_all(qcdata, function(x) as.numeric(as.character(x)))
         for (columnName in colnames(qcdata)[2:ncol(qcdata)])
             qcdata[columnName] <- as.numeric(qcdata[[columnName]])
     }
 
-    # Convert first col to rownames and transpose
-    rownames(qcdata) <- qcdata[[1]]
-    qcdata <- qcdata %>%
-        t() %>%
-        as.data.frame()
-    qcdata <- data.frame("Sample" = row.names(qcdata), qcdata, row.names = NULL)
-
-    assertthat::assert_that(all(metricNames %in% colnames(qcdata)),
-                            msg = "All of the specified metricNames must be included in the colnames of qcdata.")
-
-    # If only one plotType, apply to all plots
-    if (length(plotType) == 1) {
-        plotType <- rep(plotType, length(metricNames))
+    # If only one plotCategory, apply to all plots
+    if (length(plotCategory) == 1) {
+        plotCategory <- rep(plotCategory, length(metricNames))
     }
 
     plots <- list()
-
-    # Helper function
-    winsorize <- function(x, multiple = 3, na.rm = FALSE, ...){
-
-        med <- median(x, na.rm = na.rm)
-        y <- x - med # Median centered
-        sc <- stats::mad(y, center = 0, na.rm = na.rm, ...) * multiple
-        y[ y > sc ] <- sc
-        y[ y < -sc ] <- -sc
-        y + med
-    }
-
     for (metric in metricNames) {
         idx <- metric == metricNames
-        plot_type <- plotType[idx]
+        plot_metric <- plotCategory[idx]
 
         # Calculate mean and sd for hline yintercepts
         if (winsorize == TRUE) {
-            thisMetric <- winsorize(qcdata[,metric], na.rm = TRUE)
+            thisMetric <- winsorize.func(qcdata[,metric], na.rm = TRUE)
```

Comment:
`winsorize.func` is not a standard R function name

Reply:
Renamed to .winsorize